packages/apollo-collaboration-server/src/checks/checks.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable, Logger } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { CheckResultSnapshot } from 'apollo-mst'
+import {
+  CheckReport,
+  CheckReportDocument,
+  FeatureDocument,
+} from 'apollo-schemas'
+import { GetFeaturesOperation } from 'apollo-shared'
+import { Model } from 'mongoose'
+
+import { FeatureRangeSearchDto } from '../entity/gff3Object.dto'
+import { OperationsService } from '../operations/operations.service'
+
+@Injectable()
+export class ChecksService {
+  constructor(
+    private readonly operationsService: OperationsService,
+    @InjectModel(CheckReport.name)
+    private readonly checkReportModel: Model<CheckReportDocument>,
+  ) {}
+
+  private readonly logger = new Logger(ChecksService.name)
+
+  async checkFeature(doc: FeatureDocument): Promise<CheckResultSnapshot[]> {
+    const { _id, end, refSeq, start } = doc
+    const id = _id.toString()
+    return [
+      {
+        _id: `${id}-fake`,
+        name: 'FakeCheckResult',
+        ids: [id],
+        refSeq: refSeq.toString(),
+        start,
+        end,
+        message: `This is a fake result for feature ${id}`,
+      },
+    ]
+  }
+
+  /**
+   * Get all possible checkReports for given featureId
+   * @param id - featureId
+   * @returns - an array of checkReport -documents
+   */
+  async findByFeatureId(id: string) {
+    const checkReports = await this.checkReportModel.find({ ids: id }).exec()
+    return checkReports
```

Comment:
@kyostiebi Instead of `await`ing something and then immediately returning the result, you should return the result directly instead.

```suggestion
    return this.checkReportModel.find({ ids: id }).exec()
```

Reply:
Done